core/templates/pages/topics-and-skills-dashboard-page/topics-and-skills-dashboard-page.service.ts

```diff
@@ -0,0 +1,104 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service for topics and skills dashboard page.
+ */
+
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+
+import { DashboardFilter } from
+  'domain/topics_and_skills_dashboard/DashboardFilterObjectFactory';
+import { ITopicSummaryBackendDict } from
+  // eslint-disable-next-line max-len
+  'domain/topics_and_skills_dashboard/topics-and-skills-dashboard-backend-api.service';
+import { TopicsAndSkillsDashboardPageConstants } from
+  // eslint-disable-next-line max-len
+  'pages/topics-and-skills-dashboard-page/topics-and-skills-dashboard-page.constants';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TopicsAndSkillsDashboardPageService {
+  /**
```

Comment:
The original comment was:

> Add JSDoc with typeinfo and description of each arg. Try to do this for all new functions you add.

Please do this for other functions too.

Reply:
Sorry, I thought it's only to be done in services. Added them.